Source/WebKit/webpushd/WebPushDaemon.mm

```diff
@@ -705,6 +705,15 @@ static bool platformShouldPlaySound(const WebCore::NotificationData& data)
     [center getDeliveredNotificationsWithCompletionHandler:blockPtr.get()];
 }
 
+void WebPushDaemon::cancelNotification(PushClientConnection& connection, const WTF::UUID& notificationID)
+{
+    NSString *placeholderBundleIdentifier = platformNotificationCenterBundleIdentifier(connection);
+    RetainPtr center = adoptNS([[m_userNotificationCenterClass alloc] initWithBundleIdentifier:placeholderBundleIdentifier]);
+
+    [center removePendingNotificationRequestsWithIdentifiers:@[ (NSString *)notificationID.toString() ]];
```

Comment:
`(NSString *)notificationID.toString()` is not particularly cheap. We may want to cache the result given that it is used twice.